src/ModelContextProtocol/Protocol/Transport/StdioClientTransport.cs

```diff
@@ -99,6 +102,13 @@ public async Task ConnectAsync(CancellationToken cancellationToken = default)
             }
             _logger.TransportProcessStarted(EndpointName, _process.Id);
             _processStarted = true;
+            
+            // Create streams with explicit UTF-8 encoding to ensure proper Unicode character handling
+            // This is especially important for non-ASCII characters like Chinese text and emoji
+            var utf8Encoding = new UTF8Encoding(false); // No BOM
+            _stdInWriter = new StreamWriter(_process.StandardInput.BaseStream, utf8Encoding) { AutoFlush = true };
```

Comment:
Can this be achieved by setting StandardOutputEncoding on the ProcessStartInfo, instead of wrapping another writer around the underlying stream?